dev/TEAL_opcodes.md

```diff
@@ -730,6 +764,46 @@ When A is a uint64, index 0 is the least significant bit. Setting bit 3 to 1 on
 - pop a byte-array A, integer B, and small integer C (between 0..255). Set the Bth byte of A to C, and push the result
 - LogicSigVersion >= 3
 
+## extract s l
+
+- Opcode: 0x57 {uint8 start position} {uint8 length}
+- Pops: *... stack*, []byte
+- Pushes: []byte
+- pop a byte-array A. For immediate values in 0..255 S and L: extract a range of bytes from A starting at S up to but not including S+L, push the substring result. If L is 0, then extract to the end of the string. If S or S+L is larger than the array length, the program fails
```

Comment:
same comment as in the short version about bounds for `S` and `S+L`

also above it says `start position` / `length`
while it says `S` / `L` in the description

Reply:
This seems true throughout the doc - when describing the exact opcode in the file, we give the hex value, and then the immediates, but here we give them all descriptive names like: ` {uint8 transaction field index}` and `{uint8 position in scratch space to load from}`